source/common/common/key_value_store_base.cc

```diff
@@ -58,20 +70,31 @@ bool KeyValueStoreBase::parseContents(absl::string_view contents) {
   return true;
 }
 
-void KeyValueStoreBase::addOrUpdate(absl::string_view key_view, absl::string_view value_view) {
+void KeyValueStoreBase::addOrUpdate(absl::string_view key_view, absl::string_view value_view,
+                                    absl::optional<std::chrono::milliseconds> ttl) {
   ENVOY_BUG(!under_iterate_, "addOrUpdate under the stack of iterate");
   std::string key(key_view);
   std::string value(value_view);
+  // Remove ttl if <= 0
+  if (ttl && ttl <= std::chrono::milliseconds(0)) {
+    ASSERT(false);
+    ttl = absl::nullopt;
+  }
```

Comment:
If a TTL of 0 is an error, then we should probably not store the entry in the KV store at all. In other words, I'm suggest a return here, not setting the ttl to nullopts, I also wonder if we should enforce a min TTL.